master-cloudconfig.yaml

```diff
@@ -0,0 +1,856 @@
+      [Unit]
+      Description=Timer
+
+      [Timer]
+      OnCalendar=15:00
+      Unit=k8s-controller-manager-restart.service
+
+      [Install]
+      WantedBy=multi-user.target
+  - name: k8s-scheduler.service
+    enable: true
+    command: start
+    content: |
+      [Unit]
+      Description=k8s-scheduler Service
+      Requires=calico-node.service
+      After=calico-node.service
+
+      [Service]
+      Restart=always
+      RestartSec=0
+      TimeoutStopSec=10
+      StartLimitIntervalSec=0
+      EnvironmentFile=/etc/network-environment
+      Environment="IMAGE={{.Spec.Docker.ImageNamespace}}/hyperkube:{{.Spec.Kubernetes.Hyperkube.Version}}"
+      Environment="NAME=%p.service"
+      Environment="NETWORK_CONFIG_CONTAINER="
+      ExecStartPre=/usr/bin/docker pull $IMAGE
+      ExecStartPre=-/usr/bin/docker stop -t 10 $NAME
+      ExecStartPre=-/usr/bin/docker rm -f $NAME
+      ExecStart=/usr/bin/docker run --rm --net=host --name $NAME \
+      -v /etc/kubernetes/ssl/:/etc/kubernetes/ssl/ \
+      -v /etc/kubernetes/config/:/etc/kubernetes/config/ \
+      $IMAGE \
+      /hyperkube scheduler \
+      --master=https://{{.Spec.Kubernetes.API.Domain}}:443 \
+      --logtostderr=true \
+      --v=2 \
+      --kubeconfig=/etc/kubernetes/config/scheduler-kubeconfig.yml
+      ExecStop=-/usr/bin/docker stop -t 10 $NAME
+      ExecStopPost=-/usr/bin/docker rm -f $NAME
+  - name: k8s-scheduler-restart.service
+    enable: true
+    content: |
+      [Unit]
+      Description=k8s-scheduler-restart
+
+      [Service]
+      Type=oneshot
+      ExecStartPre=/usr/bin/systemctl stop k8s-scheduler.service
+      ExecStartPre=/usr/bin/bash -c 'while systemctl is-active --quiet k8s-scheduler.service; do sleep 1 && echo waiting for k8s-scheduler to stop; done'
+      ExecStart=/usr/bin/systemctl start k8s-scheduler.service
+
+      [Install]
+      WantedBy=multi-user.target
+  - name: k8s-scheduler-restart.timer
+    enable: true
+    command: start
+    content: |
+      [Unit]
+      Description=Timer
+
+      [Timer]
+      OnCalendar=15:00
+      Unit=k8s-scheduler-restart.service
+
+      [Install]
+      WantedBy=multi-user.target
+  - name: k8s-addons.service
+    enable: true
+    command: start
+    content: |
+      [Unit]
+      Description=Kubernetes Addons
+      Wants=k8s-scheduler.service
+      After=k8s-scheduler.service
+      [Service]
+      Type=oneshot
+      EnvironmentFile=/etc/network-environment
+      ExecStart=/opt/k8s-addons
+      [Install]
+      WantedBy=multi-user.target
+  - name: k8s-policy-controller.service
+    enable: true
+    command: start
+    content: |
+      [Unit]
+      Description=k8s-policy-controller Service
+      Wants=k8s-api-server.service
+      Requires=k8s-addons.service
+      After=k8s-addons.service
+
+      [Service]
+      Restart=always
+      EnvironmentFile=/etc/network-environment
+      Environment="IMAGE={{.Spec.Docker.ImageNamespace}}/k8s-policy-controller:v0.2.0"
```

Comment:
I know you are dealing with what we created in the past. For the record and to have something to think about, I do not like this image namespace thing. I was thinking about having a fully qualified docker image with version tag and all that crab. The problem is, that we have to have all images we are using here inside the same namespace right now. That means we cannot easily use some official image. We can only do either or. We need more flexibility here in the future. We can talk about this at some point.